src/experimental/ProbabilisticGraphicalModels/bayesian_network.jl

```diff
@@ -189,3 +189,49 @@ function evaluate(bn::BayesianNetwork)
     end
     return evaluation_env, logp
 end
+
+function evaluate_with_values(bn::BayesianNetwork, parameter_values::AbstractVector, model)
+    # Use the exact same node ordering as BUGSModel
+    bugsmodel_node_order = model.flattened_graph_node_data.sorted_nodes
```

Comment:
This is okay for now, but eventually need some rewriting, BUGSModel is just using what `topological_sort_by_dfs` returns.